test/es6/lambda1.js

```diff
@@ -311,13 +309,53 @@ var tests = [
         }
     },
     {
-        name: "Formal parameter binding and strictness",
+        name: "'arguments' and 'eval' allowed as formal parameter names in non-strict mode",
         body: function () {
-            assert.throws(function () { eval('(eval) => { }'); }, SyntaxError, "'eval' is not allowed as a lambda formal parameter name", "Invalid usage of 'eval' in strict mode");
-            assert.throws(function () { eval('(arguments) => { }'); }, SyntaxError, "'arguments' is not allowed as a lambda formal parameter name", "Invalid usage of 'arguments' in strict mode");
-            assert.throws(function () { eval('(x, x) => { }'); }, SyntaxError, "duplicate names are not allowed in a lambda formal parameter list", "Duplicate formal parameter names not allowed in strict mode");
-            assert.throws(function () { eval('x => { let x; }'); }, SyntaxError, "lambda formal parameters cause redeclaration error with function scoped let bindings of the same name", "Let/Const redeclaration");
+            var a = eval => eval + 1;
+            var b = arguments => arguments + 2;
+            var c = (eval, arguments) => eval + arguments;

```

Comment:
Yep, the formals parsing code is shared between functions and arrow functions, so the destructuring parsing ends up being independent of removing this arrow function special cased strict mode business.